olive/data/component/dataset.py

```diff
@@ -0,0 +1,92 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import torch
+from torch.utils.data import Dataset
+
+
+class BaseDataset(Dataset):
+    """
+    This class is used to define the Olive dataset which should return the data with following format:
+    1. [data, label] for supervised learning
+    2. [data] for unsupervised learning
+    The data should be a list or dict of numpy arrays or torch tensors
+    """
+
+    def __init__(self, data, label_cols=None, **kwargs):
+        """
+        This function is used to initialize the dataset
+        """
+        self.data = data
+        self.label_cols = label_cols or []
+
+    def __len__(self):
+        """
+        This function is used to return the length of the dataset
+        """
+        return len(self.data)
+
+    def __getitem__(self, index):
+        data = {k: v for k, v in self.data[index].items() if k not in self.label_cols}
+        label = self.data[index][self.label_cols[0]]
+        return data, label
+
+    def to_numpy(self):
+        """
+        This function is used to convert the dataset to numpy array
+        """
+        pass
+
+    def to_torch_tensor(self):
+        """
+        This function is used to convert the dataset to torch tensor
+        """
+        pass
+
+    def to_snap_dataset(self):
```

Comment:
typo, snpe

Reply:
https://github.com/microsoft/Olive/pull/378 Updated in this PR.